bundles/extension.rst

```diff
@@ -6,12 +6,77 @@ file used by the application but in the bundles themselves. This article
 explains how to create and load service files using the bundle directory
 structure.
 
+There are two different ways of doing it:
+
+#. :ref:`Load your services in the main bundle class <bundle-load-services-bundle-class>`:
+   this is recommended for new bundles and for bundles following the
+   :ref:`recommended directory structure <bundles-directory-structure>`;
+#. :ref:`Create an extension class to load the service configuration files <bundle-load-services-extension>`:
+   this was the traditional way of doing it, but nowadays it's only recommended for
+   bundles following the :ref:`legacy directory structure <bundles-legacy-directory-structure>`.
+
+.. _bundle-load-services-bundle-class:
+
+Loading Services Directly in your Bundle Class
+----------------------------------------------
+
+.. versionadded:: 6.1
+
+    The ``AbstractBundle`` class was introduced in Symfony 6.1.
+
+In bundles extending the :class:`Symfony\\Component\\HttpKernel\\Bundle\\AbstractBundle`
+class, you can define the :method:`Symfony\\Component\\HttpKernel\\Bundle\\AbstractBundle::loadExtension`
+method to load service definitions from configuration files::
+
+    // ...
+    use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+    use Symfony\Component\HttpKernel\Bundle\AbstractBundle;
+
+    class AcmeHelloBundle extends AbstractBundle
+    {
+        public function loadExtension(array $config, ContainerConfigurator $containerConfigurator, ContainerBuilder $containerBuilder): void
```

Comment:
Just caught this one too (match `AbstractBundle::loadExtension()` arg names)

`public function loadExtension(array $config, ContainerConfigurator $container, ContainerBuilder $builder): void`

Linter appears to be complaining about this. Pretty sure this is a false positive in this case.

Reply:
We added a rule to explicitly name them like this, not sure its a good idea anymore 🤔 

@alamirault were you involved back then with this rule?

Reply:
I haven't looked at the code for the rule - but I'm guessing it could be improved by adding a conditional to check if the variable name matches the first|second|third... "word part" of the type name

Reply:
We've already reverted most of the changes to the longer name: https://github.com/symfony/symfony-docs/pull/18299

Let's remove the rule imho